### PR TITLE
feat(dashboards): Widget Viewer style improvements and use GridEditable component

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Changes to some of the style in the Widget Viewer.
Also updates the Widget Viewer to use the `GridEditable` component to enable resizing headers.

This change also allows us to enable sort by clicking column headers, which will be completed in another PR.
Pagination is also todo

Relying on existing snapshot tests since this change is mostly visual.

image

### CODE REVIEW

File: static/app/components/modals/widgetViewerModal.tsx [hunk 1]
```diff
@@ -35,83 +42,88 @@ type Props = ModalRenderProps &
...
   const renderWidgetViewer = () => {
-    const {organization, selection, widget, location} = props;
     const api = useApi();
-    switch (widget.displayType) {
-      case DisplayType.TABLE:
-        return (
-          <TableContainer height={FULL_TABLE_HEIGHT}>
+    const columnOrder = eventView.getColumns();
+    const columnSortBy = eventView.getSorts();
+    const Query: React.ComponentType<any> =
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetViewerModal.tsx:61]: in this case, would it be better if we did:

```suggestion
          {widget.widgetType === WidgetType.ISSUE ? (
            <IssueWidgetQueries {...commonQueryProps}>
              {({transformedResults, loading}) => {
                return (
                  <GridEditable
                    isLoading={loading}
                    data={transformedResults}
                    columnOrder={columnOrder}
                    columnSortBy={columnSortBy}
                    grid={{
                      renderHeadCell: renderGridHeaderCell(props),
                      renderBodyCell: renderGridBodyCell(props),
                    }}
                    location={location}
                  />
                );
              }}
            </IssueWidgetQueries>
          ) : (
            <WidgetQueries {...commonQueryProps}>
              {({tableResults, loading}) => {
                return (
                  <GridEditable
                    isLoading={loading}
                    data={tableResults?.[0]?.data ?? []}
                    columnOrder={columnOrder}
                    columnSortBy={columnSortBy}
                    grid={{
                      renderHeadCell: renderGridHeaderCell({
                        ...props,
                        tableData: tableResults?.[0],
                      }),
                      renderBodyCell: renderGridBodyCell({
                        ...props,
                        tableData: tableResults?.[0],
                      }),
                    }}
                    location={location}
                  />
                );
              }}
            </WidgetQueries>
          )}
```

File: static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    );
+  };
+
+export const renderGridBodyCell =
+  ({location, organization, widget, tableData}: Props) =>
+  (
+    column: TableColumn<keyof TableDataRow>,
+    dataRow: TableDataRow,
+    rowIndex: number,
+    columnIndex: number
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx:82]: shouldn't we use the following types?

```suggestion
   column: GridColumnOrder<ObjectKey>,
    dataRow: Record<string, any>,
    rowIndex: number,
    columnIndex: number
```

File: static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  location: Location;
+  organization: Organization;
+  selection: PageFilters;
+  widget: Widget;
+  tableData?: TableDataWithTitle;
+};
+
+export const renderGridHeaderCell =
+  ({selection, widget, tableData}: Props) =>
+  (column: TableColumn<keyof TableDataRow>, _columnIndex: number): React.ReactNode => {
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx:39]:
```suggestion
  (column: GridColumnOrder<ObjectKey>, _columnIndex: number): React.ReactNode => {
```

🤷‍♀️

File: static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    return (
+      <SortLink
+        align={align}
+        title={title}
+        direction={currentSort ? currentSort.kind : undefined}
+        canSort={canSort}
+        generateSortLink={generateSortLink}
+      />
+    );
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx]: 👍 updated!

File: static/app/components/modals/widgetViewerModal.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal.tsx]: 👍  updated! I initially wanted to reduce the repeat code lines but I think this works fine too.

File: static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx]: 👍 updated the type for `renderGridBodyCell`

File: static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx (hunk 3 above)
Comment (edwardgou-sentry) [reply static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx:40]: I wasn't able to reduce the type of `renderGridHeaderCell` 😞 . `renderGridHeaderCell` requires a `type` attribute on the `column` arg which is not available in `GridColumnOrder`.

I've updated the type assertion for `renderGridHeaderCell` in `widgetViewerModal` to be more precise for now so we aren't using `any`.

I think we'll probably need to make some changes to `gridEditable/index` if we want to clean this up, which might be something for a different pr. Let me know what you think!

**Review by priscilawebdev — APPROVED** [review]

File: static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx:40]: I understand. Yes, we can address it in a follow-up PR....thanks for looking into that

**Pull request merged**